frontend/components/shelter/pet/add/FormComponents.tsx

```diff
@@ -313,23 +313,33 @@ const ImageGallery = ({
 	const imageUploadRef = useRef<HTMLInputElement>(null);
 	const [pickedImage, setPickedImage] = useState<File | null>(null);
 	const [cloned, setCloned] = useState([...images]);
-
+	
 	useEffect(() => {
 		if (!pickedImage) return;
-		const reader = new FileReader();
-		reader.onloadend = () => {
+		const waitForLoadedImage = (_pickedImage: File): Promise<string> => {
+			return new Promise((resolve, reject) => {
+				const reader = new FileReader();
+				reader.onloadend = () => {
+					resolve(reader.result as string)
+				};
+				reader.readAsDataURL(_pickedImage);
+			})
+		}
+		const runAsync = async () => {
+			const base64ImageString = await waitForLoadedImage(pickedImage)
 			setCloned((prev) => {
-				return [...prev, reader.result as string];
+				return [...prev, base64ImageString];
 			});
-		};
-
-		reader.readAsDataURL(pickedImage);
+			setPickedImage(null)
+		}
+		runAsync()
 	}, [pickedImage]);
 
 	useEffect(() => {
-		if (!onChange || !isEditMode) return;
+		if (!isEditMode) return;
 		onChange(cloned);
-	}, [cloned, isEditMode, onChange]);
+	// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [cloned, isEditMode]);
```

Comment:
eslint complaints since we are using `onChange`. We do not always have to obey eslint. I believe removing `onChange` from being watched fixed the infinite render loop since each render gives a new version (i.e. reference) of the function. `cloned` is local state, and `isEditMode` is a boolean so no worries there I believe